cmd/client-fs.go

```diff
@@ -212,21 +212,31 @@ func (f *fsClient) put(reader io.Reader, size int64, metadata map[string][]strin
 	// For filesystem this is a redundant information.
 
 	// Extract dir name.
-	objectDir, _ := filepath.Split(f.PathURL.Path)
-	objectPath := f.PathURL.Path
+	objectDir, objectName := filepath.Split(f.PathURL.Path)
 
-	avoidResumeUpload := isStreamFile(objectPath)
-	// Write to a temporary file "object.part.minio" before commit.
-	objectPartPath := objectPath + partSuffix
-	if avoidResumeUpload {
-		objectPartPath = objectPath
-	}
 	if objectDir != "" {
 		// Create any missing top level directories.
 		if e := os.MkdirAll(objectDir, 0777); e != nil {
 			err := f.toClientError(e, f.PathURL.Path)
 			return 0, err.Trace(f.PathURL.Path)
 		}
+		// Check if object name is empty, must be an empty directory
+		if objectName == "" {
+			reader = hookreader.NewHook(reader, progress)
+			// Discard bytes until currentOffset.
+			if _, e := io.CopyN(ioutil.Discard, reader, 0); e != nil {
```

Comment:
I am not sure that io.CopyN(.., .., 0) doesn't something tangible at all, is this really needed to fix the bug ?

Reply:
It is need to show progress bar @vadmeste